x-pack/plugins/lens/public/state_management/index.ts

```diff
@@ -89,15 +88,6 @@ export const makeConfigureStore = (
   if (customMiddleware) {
     middleware.push(customMiddleware);
   }
-  if (process.env.NODE_ENV === 'development') {
```

Comment:
@mbondyra you should probably also remove the documentation related to `ELASTIC_LENS_LOGGER` since no longer used in `x-pack/plugins/lens/readme.md`